### ISSUE TITLE
Flood of HomeKit Alarm Notifications After Manual Mode Change

### ISSUE THREAD

**Issue by danielraffel** [haywirecoder/homebridge-envisalink-ademco#60]
I’ve recently noticed an issue where I’m getting flooded with mode change alarm state push notifications from HomeKit, even though the Homebridge logs only show one mode change. This seems to be occurring at night which aligns with the only time I really manually adjust the alarm I suppose, and the only fix I’ve found is to restart Homebridge and reboot the Envisalink Duo device. 

Have you encountered this issue before or have any suggestions for resolving it? It only seems to occur after manually adjusting the mode change switch exposed by the Envisalink Ademco plugin in the Home app. It’s happened two nights in a row, resulting in everyone in the house receiving dozens and dozens of push notifications in a short period of time.

Homebridge 1.8.5  | UI 4.6.5
Node.js v22.12.0
macOS sequoia 15.2 
Envisakink ademco plugin 2.0.15

Envisalink Duo Firmware Version: 01.00.100A

Def keeps folks on their toes!
<img width="200" alt="image" src="https://github.com/user-attachments/assets/1b5718b8-77b9-41a1-bbaf-e43992b45a17" />

**Comment by haywirecoder** [comment]
Hmm... I never encountered this prior. A few questions/recommendation
1. From what mode are you going from? Disarm to Stay? Disarm to Night?
2. Are you indicating that this happens when using the home's alarm control (tile)?
3. Does this happen immediately, after changing the alarm mode? If that is the case, I would turn on debug mode in Homebridge this will provide a lot of data about what the plug-in is receiving.

**Comment by danielraffel** [comment]
**What mode are you switching from? Disarm to Stay? Disarm to Night?**
• Disarm to Night. This morning, I noticed the issue when I switched from Night to Off, then from Off to Night, and back to Off. I couldn’t get it to trigger, so at the very least, the behavior seems non-deterministic.

**Are you saying this happens when using the alarm control tile in the Home app?**
• Yes. 
![image](https://github.com/user-attachments/assets/bb314d71-c01a-4d99-9874-041c12cc8ecc)

**Does this happen immediately after changing the alarm mode? If so, I recommend turning on debug mode in Homebridge. This will provide detailed logs of what the plugin is receiving.**
• Thanks, I’ll do that! It’s worth mentioning that I’m running Homebridge on a pretty heavily tasked Intel Mac mini. I’m wondering if this could be a memory race condition that occurs when Homebridge hasn’t been restarted for a while.

Just dawned on me that for over a year, I had a cron job restarting Homebridge hourly during daylight hours to keep local devices connected more reliably. I found some wemo WiFi lights would disconnect if I didn’t do this. Ironically, I stopped doing the restarting just a day or so ago which is also when this behavior started. 

I only changed it because when Homebridge restarts, it seems to sync the current mode to my Apple TV, and at night while watching a movie, I got tired of seeing the brief pop-up and thought I could skip the hourly restarts. But this is def worse.

I’ll enable debug mode, skip the hourly restarts until tomorrow, and check the logs tonight. Then I’ll re-enable the hourly restarts tomorrow, with a break during evening hours when I’m watching TV. My sense is that the Mac mini is overworked, gets “cranky”, and starts having issues.

**Comment by haywirecoder** [comment]
I would also review your panel for night mode. Normally, the NIGHT is "Arm-Instant (Zero Delay-Stay)," similar to the STAY mode but without the entry delay feature. It is usually associated with NIGHT STAY, which doesn't get reported correctly (just report as stay), so the plug-in has to use virtual key panel text as an indicator of night mode. So, I would also confirm that your keypad is reading night and not flipping back and forth...

Haven't had personal experience with Wemo devices, but anytime you have to restart something hourly, it bad sign that the fundamentals of the integration are working correctly.

**Comment by danielraffel** [comment]
Since I stopped restarting Homebridge hourly I haven’t experienced any issues. I also verified that Night is correctly setting Night-Stay on the panel without flipping states (and shows as Night in the Home app.) I’m attributing the push issues to an overworked, older machine that was likely overloaded and didn’t benefit from such frequent restarts. Thanks so much for all the support. Your plugin has been a very huge quality of life upgrade for my family. 🙇

![image](https://github.com/user-attachments/assets/4c426762-0e8f-4f7f-9da0-eb61817fc95b)
![image](https://github.com/user-attachments/assets/441001fc-f4b2-4b61-8629-cb9953d39113)
